src/Codeception/Lib/InnerBrowser.php

```diff
@@ -834,10 +835,14 @@ protected function proceedSubmitForm(Crawler $frmCrawl, array $params, $button =
             ));
             if (count($btnCrawl)) {
                 $requestParams[$button] = $btnCrawl->attr('value');
+                $formaction = $btnCrawl->attr('formaction');
+                if ($formaction) {
+                    $url = $formaction;
+                }
             }
         }
 
-        $url = $this->getFormUrl($frmCrawl);
+        $url = $url ? $url : $this->getFormUrl($frmCrawl);
```

Comment:
This expression could be shortened to `$url = $url ?: $this->getFormUrl($frmCrawl);`,
But I think that using `if` is a better style:
```php
if (!$url) {
    $url = $this->getFormUrl($frmCrawl);
}
```